### ISSUE TITLE
The calculation of `max_string_length` causes horrible load times on certain Safari versions, and there must be a more efficient way to do this.

### ISSUE THREAD

**Issue by DoctorBud** [brython-dev/brython#1694]
I noticed that on certain newer versions of Safari on macOS, Brython takes *minutes* to load. I eventually tracked the problem down to an inefficient algorithm used during library load here:

- https://github.com/brython-dev/brython/commit/bcefc652a9fd7d7c161220330f3f16f7fb33ed2f#diff-8e8ecca0b3db71218a6db407a96c2d18616dd7c0580602cac7c8faf08565aae8R63

The code is using a binary search algorithm to repeatedly try to allocate very large strings, and then to catch the exceptions that are thrown until an upper bound is achieved. This appears to work quickly on Chrome/Firefox and older versions of Safari. But in the versions of Safari I'm using Brython with, there is a problem, and the repeated allocation/exception takes minutes.

I've created a simple example of this algorithm in a standalone HTML file here, as well as some additional documentation.

[BrythonBug.html.zip](https://github.com/brython-dev/brython/files/6645763/BrythonBug.html.zip)

The source for the above looks like:

```
<!DOCTYPE HTML>
<html>

  <body>
    <p>
      This example HTML file demonstrates that Safari Version 14.0.3 (15610.4.3.1.7, 15610) on MacOS Catalina and Safari Technology Preview Release 123 (Safari 14.2, WebKit 15612.1.7.10) have serious performance problems with String.repeat(n).
    </p>

    <p>
      Brython recently added a commit
      <a href="https://github.com/brython-dev/brython/commit/bcefc652a9fd7d7c161220330f3f16f7fb33ed2f">Add attribute "max_string_lenth" to sys (Brython-specific; Javascript string have a maximum length, different between browsers).
      </a>
    </p>
    <p>
      The implementation of calculating the max string length relies upon trying (and failing) to allocate long strings until it eventually decides the max string length for the current browser platform. Unfortunately, this heavy-handed approach takes a significant amount of time and energy on macOS Safari. Firefox and Chrome appear not to notice, and perform the computations quickly.
    </p>

    <p>
      My recommendation to the Brython team is to come up with a more reasonable way to compute max string length. Perhaps just a set of platform-specific constants that are hardcoded or derived periodically via a script.
    </p>
    <script>
      var nb=Math.pow(2,29);
      while(true) {
        try{
          console.log('repeat nb', nb);
          'a'.repeat(nb);
          nb*=2;
        }
        catch(err) {
          var lo=nb/2;
          var hi=nb;
          var mid;
          while (hi-lo > 1) {
            try {
              mid=lo+(hi-lo)/2;
              console.log('repeat mid', mid);
              'a'.repeat(mid);
              lo=mid;
            }
            catch(err) {
              hi=mid;
            }
          }
          try {
            console.log('repeat hi', hi);
            'a'.repeat(hi);
            console.log(`max_string_length hi: ${hi}`);
            break;
          }
          catch(err) {
            console.log(`max_string_length lo: ${lo}`);
            break;
          }
        }
      }
    </script>
    <p>...After the script.</p>
  </body>

</html>
```

**Comment by PierreQuentel** [comment]
I suppose that you are using version 3.9.2 and that it's the same problem as reported in issue #1663 ? If so it was fixed a few weeks ago and should not happen with version 3.9.3.

**Comment by PierreQuentel** [comment]
@DoctorBud 

In issue #1695 Andy Lewis reported that Brython still crashed on iPads with version 3.9.3 loaded from a CDN.
This should be fixed with the new 3.9.4 release.
Can you confirm ?

**Comment by PierreQuentel** [comment]
I think the bug is fixed, so I close the issue. If it's not the case, feel free to reopen it.

Thanks again !
